lib/repl.js

```diff
@@ -314,40 +308,42 @@ function REPLServer(prompt,
       debug('finish', e, ret);
       self.memory(cmd);
 
-      if (e && !self.bufferedCommand && cmd.trim().match(/^npm /)) {
```

Comment:
We always shortcircuit if there is an error; group with the rest of the error handling.